implementations/rust/ockam/ockam_entity/src/worker/create_key.rs

```diff
@@ -69,23 +70,19 @@ impl CreateKeyChange {
     }
 }
 
-impl<V: ProfileVault> ProfileImpl<V> {
-    /// Create a new key event
-    pub(crate) fn create_key_event_static(
+impl ProfileState {
+    /// Create a new key
+    pub fn create_key_static(
         prev_id: EventIdentifier,
         key_attributes: KeyAttributes,
-        attributes: Option<ProfileEventAttributes>,
+        attributes: ProfileEventAttributes,
         root_key: Option<&Secret>,
-        vault: &mut V,
+        vault: &mut VaultSync,
     ) -> ockam_core::Result<ProfileChangeEvent> {
-        let attributes = attributes.unwrap_or_default();
-
-        // TODO: Should be customisable
-        let secret_attributes = SecretAttributes::new(
-            SecretType::Curve25519,
-            SecretPersistence::Persistent,
-            CURVE25519_SECRET_LENGTH,
-        );
+        let secret_attributes = match key_attributes.meta() {
+            MetaKeyAttributes::SecretAttributes(secret_attributes) => *secret_attributes,
```

Comment:
I'm not sure this the mechanism we use for adjusting secret type

Reply:
Notes from our conversation: Talking about using labels and different kinds of secrets. Users will not not when they will need to create things like BLS secret vs other secret. Consider APIs using predefined label such as "signing key" and API maybe like `create_signing_key` etc for all of the various kinds of secrets.